org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIDebugTarget.java

```diff
@@ -1340,6 +1344,13 @@ protected void resume(boolean fireNotification) throws DebugException {
 			resumeThreads();
 			VirtualMachine vm = getVM();
 			if (vm != null) {
+				if (pendingResumeRequests) {
+					while (suspendCount > 1) {
+						vm.resume();
+						suspendCount--;
+						pendingResumeRequests = false;
+					}
+				}
```

Comment:
Shouldn't `vm.resume();` call below be in the `else` block?

Reply:
This is a default `vm.resume();` which was existing.
When the `suspendCount  = 1` this works by default.
If there were more than 1 Suspend VM breakpoints are set they will be resumed inside while loop.
Whenever a `IJavaBreakpoint.SUSPEND_VM` breakpoint is hit `JDIDebugTarget#suspendedByBreakpoint(JavaBreakpoint breakpoint, boolean queueEvent, EventSet set)` is invoked.
So I am counting the number of Suspend VM breakpoints hit. And whenever VM(Debug Target) Resume is requested all the pending VM suspend will be cleared inside while loop.